src/main/java/com/aws/greengrass/util/GreengrassServiceClientFactory.java

```diff
@@ -87,12 +94,25 @@ private boolean validString(Node node, String key) {
         return node != null && node.childOf(key) && Utils.isNotEmpty(Coerce.toString(node));
     }
 
+    /**
+     * Retrieve configValidationError.
+     * Validate again if the device config has changed.
+     *
+     */
+    public String getConfigValidationError() {
```

Comment:
make this Optional<String> or add a javadoc that this will return `null` if there is no error

Reply:
I checked that only `GreengrassRepositoryDownloader::checkDownloadable` calls `getConfigValidationError`. Will make this optional and modify the usage.